modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/components/SidebarPanelInfoView/VersionsContent.tsx

```diff
@@ -32,42 +32,49 @@ const VersionsContent = ({
 	const [totalPages, setTotalPages] = useState(1);
 	const [versions, setVersions] = useState([] as IVersion[]);
 
-	useEffect((): void => {
-		const getVersionsData = async (): Promise<void> => {
-			try {
-				setLoading(true);
+	const getVersionsData = useCallback(async (): Promise<void> => {
+		try {
+			setLoading(true);
+
+			const response: Response = await fetch(
+				`${getItemVersionsURL}&${namespace}pageNumber=${currentPage}`
+			);
 
-				const response: Response = await fetch(
-					`${getItemVersionsURL}&${namespace}pageNumber=${currentPage}`
+			if (!response.ok) {
+				throw new Error(
+					`Failed to fetch ${getItemVersionsURL}&${namespace}pageNumber=${currentPage}`
 				);
+			}
 
-				if (!response.ok) {
-					throw new Error(
-						`Failed to fetch ${getItemVersionsURL}&${namespace}pageNumber=${currentPage}`
-					);
-				}
+			const {totalPages, versions}: IData = await response.json();
 
-				const {totalPages, versions}: IData = await response.json();
+			setVersions(versions);
+			setTotalPages(totalPages);
+		}
+		catch (error: unknown) {
+			onError();
 
-				setVersions(versions);
-				setTotalPages(totalPages);
+			if (process.env.NODE_ENV === 'development') {
+				console.error(error);
 			}
-			catch (error: unknown) {
-				onError();
+		}
+		finally {
+			setLoading(false);
+		}
+	}, [currentPage, getItemVersionsURL, namespace, onError]);
 
-				if (process.env.NODE_ENV === 'development') {
-					console.error(error);
-				}
-			}
-			finally {
-				setLoading(false);
-			}
-		};
+	useEffect((): void => {
```

Comment:
Hi @claraizquierdo !
After giving some thoughts to this this problem, I think we can simplify just using the `useIsFirstRender` hook and cancelling the fetch in that case, this way:

```
	useEffect(() => {
		if (isFirst) {
			return;
		}
		getVersionsData();
	}, [getVersionsData, isFirst]);
```

If I'm not mistaken, this way we avoid the initial unnecessary fetch, but get the data on first tab and page changes.
I'll push some changes.
Do you mind to give it a try?
Thxs!